src/app/page.tsx

```diff
@@ -1,11 +1,59 @@
 import { getServerAuthSession } from "~/server/auth";
 import Users from "~/app/users/page";
+import Wizard from "~/components/organisms/wizard/Wizard";
 
 const Home = async () => {
   const session = await getServerAuthSession();
   if (!session?.user) return null;
-
-  return <Users />;
+  const inputs = [
+    [
+      {
+        type: "text",
+        placeholder: "Your username goes here",
+        label: "Username",
+      },
+      {
+        type: "password",
+        placeholder: "Your password goes here",
+        label: "Password",
+      },
+      {
+        type: "password",
+        placeholder: "Repeat your password here",
+        label: "Repeat Password",
+      },
+    ],
+    [
+      {
+        type: "email",
+        label: "Email Address",
+        placeholder: "Your email goes here",
+      },
+      {
+        type: "text",
+        label: "Phone Number",
+        placeholder: "Your phone number goes here",
+      },
+    ],
+    [
+      {
+        type: "text",
+        label: "Credit Card Number",
+        placeholder: "Your credit card number goes here",
+      },
+      {
+        type: "text",
+        label: "Billing Address",
+        placeholder: "Your billing address goes here",
+      },
+    ],
+  ];
+  return (
+    <>
+      <Users />
+      <Wizard steps={["Sign Up", "Details", "Billing"]} inputs={inputs} />
```

Comment:
This approach can cause multiple issues. 
Developer can define `steps` array with one length and `inputs` array with different length which can break the `Wizard` component. 
I can see two solutions for this.
1. This one can be easier to implement is to define inputs as `Record<string, WizardInputValue`. Like that input will have both the keys and values which need to be passed to the `Wizard` component. (Example at the end)
2. Create validation for the current implementation that typescript is complaining at compile time if `steps.length != inputs.length`


Example for solution one
```typescript
type WizardInputValue = {
  type: 'text' | 'number' | 'data' | 'select' | 'checkbox',
  label: string,
  placeholder: string
}

interface WizardProps {
  inputs: Record<string, WizardInputValue>
}
```

Reply:
I prefer the first approach as it prevents the initial problem rather than fixing it. But I don't think `Record` is a suitable type as we need our steps to be ordered. Here I used an array instead:

```ts
interface Fields {
  type: "text" | "password" | "email" | "number" | "tel" | "url" | "search";
  label: string;
  placeholder: string;
}

interface FormStep {
  name: string;
  fields: Fields[];
}

interface FormProps {
  form: FormStep[];
}
```
What do you think?

Reply:
I like it 🚀

Reply:
I like it as well, but one thing that I would like to add as a suggestion is that we add a `name` property to the `Fields` type as well since the `name` is the main part that is being used to know which field is which when submitting a form.

Reply:
Also, one thing I remembered, but this approach could cause us some issues with the ability to add values to a `select` element if we're required to add e.g. "Select your country" to the form.